packages/eslint-plugin-gestalt/src/eslintASTHelpers.js

```diff
@@ -0,0 +1,313 @@
+*/
+
+type BuildPropsFromKeyValuesType = ({|
+  keyValues: GenericType,
+|}) => string;
+
+export const buildPropsFromKeyValues: BuildPropsFromKeyValuesType = ({ keyValues }) => {
+  const newKeyValues = [...keyValues];
+  const stringProps = newKeyValues
+    .sort((a, b) => {
+      if (a.key < b.key) return -1;
+      if (a.key > b.key) return 1;
+      return 0;
+    })
+    .map(
+      (pair) =>
+        `${pair.key}=${
+          pair.isValueTypeLiteral
+            ? buildLiteralValueString({ value: pair.value })
+            : `{${pair.value}}`
+        }`,
+    )
+    .join(' ');
+
+  return stringProps;
+};
+
+/* ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓  buildPropsFromKeyValuesVariable ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓
+This function returns key/values formatted as component props.
+Examples:
+[{ key: "color", value: "red", isValueTypeLiteral: true }, { key: "width", value: 20, isValueTypeLiteral: true }] >> 'color="red" width={20}'
+*/
+type BuildPropsFromKeyValuesVariableType = ({|
+  context: GenericType,
+  variableNode: GenericType,
+|}) => string;
+
+export const buildPropsFromKeyValuesVariable: BuildPropsFromKeyValuesVariableType = ({
+  context,
+  variableNode,
+}) => {
+  const keyValues = retrieveKeyValuesFromVariable({
+    context,
+    variableNode,
+  });
+
+  return buildPropsFromKeyValues({ keyValues });
+};
+
+/* ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓  getComponenFromAttribute ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓
+This function returns the component containing the attribute's node (nodeAttribute).
+Examples:
+<div {...props} /> >> returns div node for the spread props attribute
+*/
+type GetComponentFromAttributeType = ({| nodeAttribute: GenericType |}) => GenericType;
+
+export const getComponenFromAttribute: GetComponentFromAttributeType = ({ nodeAttribute }) =>
+  nodeAttribute.parent;
+
+/* ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓  getComponentNameFromAttribute ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓
+This function returns the component's name containing the attribute's node (nodeAttribute).
+Examples:
+<div {...props} /> >> returns div for the spread props attribute
+*/
+type GetVariableNodeInScopeFromNameType = ({|
+  context: GenericType,
+  nodeElement: GenericType,
+  name: string,
+|}) => GenericType;
+
+export const getVariableNodeInScopeFromName: GetVariableNodeInScopeFromNameType = ({
+  context,
+  nodeElement,
+  name,
+}) => {
+  const scope = context.getScope(nodeElement);
+  // Look in local scope for variable reference
+  const variableNode = scope.references.find((reference) => reference.identifier.name === name);
+  return variableNode;
+};
+
+/* ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓  getComponentNameFromAttribute ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓
+This function returns the component's name containing the attribute's node (nodeAttribute).
+Examples:
+<div {...props} /> >> returns div for the spread props attribute
+*/
+type GetComponentNameFromAttributeType = ({| nodeAttribute: GenericType |}) => string;
+
+export const getComponentNameFromAttribute: GetComponentNameFromAttributeType = ({
+  nodeAttribute,
+}) => nodeAttribute.parent.name.name;
```

Comment:
Do we trust that this path will always be valid? I'm so apprehensive about these deep dives when we don't have Flow to help us out 😬 I've been being overly-defensive with optional chaining to compensate

Reply:
ADDED  optional chaining
DONE